smack-im/src/main/java/org/jivesoftware/smack/roster/Roster.java

```diff
@@ -469,11 +470,14 @@ public void reload() throws NotLoggedInException, NotConnectedException, Interru
             @Override
             public void processException(Exception exception) {
                 rosterState = RosterState.uninitialized;
-                Level logLevel;
+                Level logLevel = Level.SEVERE;
                 if (exception instanceof NotConnectedException) {
                     logLevel = Level.FINE;
-                } else {
-                    logLevel = Level.SEVERE;
+                } else if (exception instanceof XMPPErrorException) {
+                    StanzaError error = ((XMPPErrorException) exception).getStanzaError();
+                    if (error.getCondition() == Condition.feature_not_implemented) {
```

Comment:
It's a common error to assume that a server should return [feature-not-implemented](https://datatracker.ietf.org/doc/html/rfc6120#page-111) when a feature - here: the XMPP roster - is not implemented. However, if you carefully read RFC 6120, then you may notice that it probably should be [service-unavailable](https://datatracker.ietf.org/doc/html/rfc6120#page-121). I'd say we simply check for both.

Reply:
Done.